routing/astar_router.cpp

```diff
@@ -0,0 +1,184 @@
+#include "astar_router.hpp"
+
+#include "routing/base/astar_algorithm.hpp"
+#include "routing/base/astar_progress.hpp"
+#include "routing/bicycle_directions.hpp"
+#include "routing/bicycle_model.hpp"
+#include "routing/car_model.hpp"
+#include "routing/features_road_graph.hpp"
+#include "routing/index_graph.hpp"
+#include "routing/pedestrian_model.hpp"
+#include "routing/route.hpp"
+#include "routing/turns_generator.hpp"
+
+#include "indexer/feature_altitude.hpp"
+#include "indexer/routing_section.hpp"
+
+#include "geometry/distance.hpp"
+#include "geometry/point2d.hpp"
+
+namespace
+{
+size_t constexpr kMaxRoadCandidates = 6;
+float constexpr kProgressInterval = 2;
+
+using namespace routing;
+
+vector<Junction> ConvertToJunctions(IndexGraph const & graph, vector<JointId> const & joints)
+{
+  vector<FSegId> const & fsegs = graph.RedressRoute(joints);
+
+  vector<Junction> junctions;
+  junctions.reserve(fsegs.size());
+
+  Geometry const & geometry = graph.GetGeometry();
+  for (FSegId const & fseg : fsegs)
+    junctions.emplace_back(geometry.GetPoint(fseg), feature::kDefaultAltitudeMeters);
+
+  return junctions;
+}
+}  // namespace
+
+namespace routing
+{
+AStarRouter::AStarRouter(Index const & index, TCountryFileFn const & countryFileFn,
+                         shared_ptr<VehicleModelFactory> vehicleModelFactory,
+                         unique_ptr<IDirectionsEngine> directionsEngine)
+  : m_index(index)
+  , m_countryFileFn(countryFileFn)
+  , m_roadGraph(
+        make_shared<FeaturesRoadGraph>(index, IRoadGraph::Mode::ObeyOnewayTag, vehicleModelFactory))
+  , m_vehicleModelFactory(vehicleModelFactory)
+  , m_directionsEngine(move(directionsEngine))
+{
+}
+
+string AStarRouter::GetName() const { return "AStar"; }
```

Comment:
Let's name the router according to our style. Please look at road_graph_router.cpp. I offer: "astar-bidirectional-car"